provider/vsphere/config.go

```diff
@@ -107,6 +119,19 @@ func (c environConfig) validate() error {
 			return errors.Errorf("%s: must not be empty", field)
 		}
 	}
+
+	if diskProvType, ok := c.attrs[cfgDiskProvisioningType]; ok {
+		found := false
+		for _, val := range vsphereclient.ValidDiskProvisioningTypes {
+			if vsphereclient.DiskProvisioningType(diskProvType.(string)) == val {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return errors.Errorf("invalid disk provisioning type %s", diskProvType)
```

Comment:
change to
`return errors.Errorf("disk provisioning type %q must be one of %q", diskProvType, vsphereclient.ValidDiskProvisioningTypes)`

Reply:
This would result in something like:

```bash
invalid base config: disk provisioning type "eroneous" must be one of ["thin" "thick" "thickEagerZero"]
```
which sounds a bit confusing.

maybe something like:

```go
return errors.Errorf("%q must be one of %q", cfgDiskProvisioningType, vsphereclient.ValidDiskProvisioningTypes)
```

which would look like:

```bash
invalid base config: "disk-provisioning-type" must be one of ["thin" "thick" "thickEagerZero"]
```

Thoughts?

Reply:
I'm happy with the current version:

ERROR invalid config change: "disk-provisioning-type" must be one of ["thin" "thick" "thickLazyZero"]